test/mpi/main.cc

```diff
@@ -16,8 +16,10 @@ static std::vector<ucc_coll_type_t> colls = {
     UCC_COLL_TYPE_SCATTER,        UCC_COLL_TYPE_SCATTERV};
 static std::vector<ucc_coll_type_t> onesided_colls = {UCC_COLL_TYPE_ALLTOALL};
 static std::vector<ucc_memory_type_t> mtypes = {UCC_MEMORY_TYPE_HOST};
-static std::vector<ucc_datatype_t> dtypes = {UCC_DT_INT32, UCC_DT_INT64,
-                                             UCC_DT_FLOAT32, UCC_DT_FLOAT64};
+static std::vector<ucc_datatype_t>    dtypes         = {
+    UCC_DT_INT32,           UCC_DT_INT64,           UCC_DT_FLOAT32,
+    UCC_DT_FLOAT64,         UCC_DT_FLOAT128,        UCC_DT_FLOAT32_COMPLEX,
```

Comment:
this dtypes is the list of types test by default. lets not make it super large since it will slow down all tests by default. lets keep int32,64,float32,64,and float64_complex . user can always run for extended set.

Reply:
done